frontend/src/components/Psychologist/Reimbursement/Bill.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from 'react';
+
+import Logo from 'components/Logo/LogoHeader';
+
+import agent from 'services/agent';
+import { formatFrenchDate, formatMonth } from 'services/date';
+
+import { useParams } from 'react-router-dom';
+import BillingTable from './BillingTable';
+
+import styles from './bill.cssmodule.scss';
+
+const Bill = () => {
+  const { month, year } = useParams();
+  const [user, setUser] = useState({});
+  const [appointments, setAppointments] = useState([]);
+
+  useEffect(() => {
+    agent.Appointment.get().then(response => {
+      const appointmentsByDate = {};
+      response.appointments.forEach(appointment => {
+        const existingValue = appointmentsByDate[appointment.appointmentDate];
+        appointmentsByDate[appointment.appointmentDate] = existingValue ? existingValue + 1 : 1;
+      });
+      setAppointments(appointmentsByDate);
+    });
+    agent.User.getConnected().then(response => {
+      setUser(response.data);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (Object.keys(appointments).length > 0 && Object.keys(user).length > 0) {
```

Comment:
Why `Object.keys(user).length > 0`  ? `user` contains connected user data, right ? so checking that it exists should be enough, isn't it?

Also, is it necessary to call `agent.User.getConnected()` explicitely here? Isn't this information available in the userStore?

Reply:
it's open in a new page so no it's not somewhere else
and I initialize both this object to {} (to avoid null ref) so I need this :/